cmd/wallet-web/src/components/DIDManagement/ImportDid.vue

```diff
@@ -0,0 +1,158 @@
+    </div>
+
+    <button id="saveDIDBtn" class="mt-5 btn-primary" @click="saveAnyDID">
+      Resolve and Save Digital Identity
+    </button>
+
+    <div v-if="saveErrors.length">
+      <b>Please correct the following error(s):</b>
+      <ul>
+        <li v-for="error in saveErrors" :key="error">{{ error }}</li>
+      </ul>
+    </div>
+
+    <div v-if="saveAnyDIDSuccess">
+      <div class="">
+        <label id="save-anydid-success">Saved your DID successfully.</label>
+      </div>
+    </div>
+
+    <div>
+      <span class="h-80">{{ anyDidDocTextArea }}</span>
+    </div>
+  </div>
+</template>
+
+<script>
+import { DIDManager } from '@trustbloc/wallet-sdk';
+import { mapGetters } from 'vuex';
+
+export default {
+  name: 'ImportDid',
+  data() {
+    return {
+      anyDidDocTextArea: '',
+      keyType: 'ED25519',
+      didID: '',
+      privateKeyStr: '',
+      keyID: '',
+      saveErrors: [],
+      loading: false,
+      saveAnyDIDSuccess: false,
+      importKeyType: '',
+      keyFormat: '',
+    };
+  },
+  computed: {
+    showImportKeyType() {
+      return this.keyFormat == 'Base58';
+    },
+  },
+  created: function () {
+    const agent = this.getAgentInstance();
+    const { user } = this.getCurrentUser().profile;
+    this.didManager = new DIDManager({ agent, user });
+  },
+  methods: {
+    ...mapGetters('agent', { getAgentInstance: 'getInstance' }),
+    ...mapGetters(['getCurrentUser']),
+    saveAnyDID: async function () {
+      this.saveErrors.length = 0;
+      this.saveAnyDIDSuccess = false;
+      this.anyDidDocTextArea = '';
+      this.loading = true;
+
+      if (this.didID.length == 0) {
+        this.saveErrors.push('did id required.');
+        return;
+      }
+
+      if (this.keyFormat.length == 0) {
+        this.saveErrors.push('please select format of the key being imported.');
+        return;
+      }
+
+      if (this.privateKeyStr.length == 0) {
+        this.saveErrors.push('private key is required.');
+        return;
+      }
+
+      if (this.keyID.length == 0) {
+        this.saveErrors.push('key ID (verification method) matching private key is required.');
+        return;
+      }
+
+      if (this.keyFormat == 'Base58' && this.importKeyType.length == 0) {
+        this.saveErrors.push('key type of private key for importing base58 private keys.');
+        return;
+      }
+
+      try {
+        await this.didManager.importDID(this.getCurrentUser().profile.token, {
+          did: this.didID,
+          key: {
+            keyType: this.importKeyType,
+            privateKeyBase58: this.keyFormat == 'Base58' ? this.privateKeyStr : '',
+            privateKeyJwk: this.keyFormat == 'JWK' ? JSON.parse(this.privateKeyStr) : undefined,
+            keyID: this.keyID,
+          },
+        });
+      } catch (e) {
+        this.loading = false;
+        this.anyDidDocTextArea = `failed to import did: ${e.toString()}`;
+        return;
+      }
+
+      this.loading = false;
```

Comment:
did you miss to emit event here to update did list?

Reply:
Updated